lib/ts/recipe/thirdpartypasswordless/recipe.tsx

```diff
@@ -202,24 +188,42 @@ export default class ThirdPartyPasswordless extends AuthRecipe<
      * Static methods.
      */
 
-    static init(
-        config: UserInput
-    ): CreateRecipeFunction<GetRedirectionURLContext, PreAndPostAPIHookAction, OnHandleEventContext, NormalisedConfig> {
-        return (
-            appInfo: NormalisedAppInfo
-        ): RecipeModule<GetRedirectionURLContext, PreAndPostAPIHookAction, OnHandleEventContext, NormalisedConfig> => {
-            ThirdPartyPasswordless.instance = new ThirdPartyPasswordless(
-                {
-                    ...config,
-                    appInfo,
-                    recipeId: ThirdPartyPasswordless.RECIPE_ID,
+    static init(config: UserInput): InitOutput {
+        return {
+            authReact: (
+                appInfo: NormalisedAppInfo
+            ): RecipeModule<
+                GetRedirectionURLContext,
+                PreAndPostAPIHookAction,
+                OnHandleEventContext,
+                NormalisedConfig
+            > => {
+                ThirdPartyPasswordless.instance = new ThirdPartyPasswordless(
+                    {
+                        ...config,
+                        appInfo,
+                        recipeId: ThirdPartyPasswordless.RECIPE_ID,
+                    },
+                    {
+                        passwordlessInstance: undefined,
+                        thirdPartyInstance: undefined,
+                    }
+                );
+                return ThirdPartyPasswordless.instance;
+            },
+            webJS: ThirdpartyPasswordlessWebJS.init({
+                ...config,
+                override: {
+                    functions: (originalImpl, builder) => {
+                        const functions = getFunctionOverrides(ThirdPartyPasswordless.RECIPE_ID, config?.onHandleEvent);
```

Comment:
would not we be calling `normaliseThirdPartyPasswordlessConfig` twice if we call it inside `init` here ? normalisation requires appInfo and it is not there yet

Reply:
also `normaliseThirdPartyPasswordlessConfig` requires `appInfo` but it is only available in `authReact` thing

Reply:
- if the call was moved here instead of the constructor, and the constructor took the normalised config it'd be only called once.
- the appInfo (and the recipeId) passes through the normalisation unchanged.